File: Habitica/src/com/magicmicky/habitrpgwrapper/lib/models/QuestProgress.java
package com.magicmicky.habitrpgwrapper.lib.models;

import com.habitrpg.android.habitica.HabitDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

import java.util.HashMap;

/**
 * Created by viirus on 06/07/15.
 */
@Table(databaseName = HabitDatabase.NAME)
public class QuestProgress extends BaseModel {

    @Column
    @PrimaryKey(autoincrement = true)
    long id;

    @Column
    private float down, up;

    @Column
    public double hp, rage;

    public HashMap<String, Integer> collect;

    private QuestProgress(float down, float up) {
        this.down = down;
        this.up = up;
    }

    public float getDown() {
        return down;
    }

    public void setDown(float down) {
        this.down = down;
    }

    public float getUp() {
        return up;
    }

    public void setUp(float up) {
        this.up = up;
    }

    public QuestProgress() {}
}

File: shared/src/androidMain/kotlin/com/habitrpg/shared/habitica/models/inventory/QuestProgress.kt
package com.habitrpg.shared.habitica.models.inventory

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

actual open class QuestProgress : RealmObject() {

    @PrimaryKey
    actual var id: String? = null
    actual var key: String? = null
    actual var hp: Double = 0.0
    actual var rage: Double = 0.0
    actual var collect: RealmList<QuestProgressCollect>? = null
    actual var down: Float = 0.0f
    actual var up: Float = 0.0f
}
